### PR TITLE
fix for #4364

### LINKED ISSUE #4364: Bug in mne.connectivity.spectral_connectivity()?
Hi all,

I was just trying to use `mne.connectivity.spectral_connectivity()` in the following way:

```python
# perform the connectivity analysis:
conn = mne.connectivity.spectral_connectivity(epochs, method='wpli',
                                              fmin=1, fmax=40, n_jobs=6)
```

I get the following bug:

<details><summary>Error Trace, click to expand</summary>

```
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-3-56a8545dd948> in <module>()
     56     # perform the connectivity analysis:
     57     conn = mne.connectivity.spectral_connectivity(epochs, method='wpli',
---> 58                                                   fmin=1, fmax=40, n_jobs=6)
     59 

/Library/anaconda/envs/py36/lib/python3.6/site-packages/mne/connectivity/spectral.py in spectral_connectivity(data, method, indices, sfreq, mode, fmin, fmax, fskip, faverage, tmin, tmax, mt_bandwidth, mt_adaptive, mt_low_bias, cwt_frequencies, cwt_n_cycles, block_size, n_jobs, verbose)

/Library/anaconda/envs/py36/lib/python3.6/site-packages/mne/utils.py in verbose(function, *args, **kwargs)
    724         with use_log_level(verbose_level):
    725             return function(*args, **kwargs)
--> 726     return function(*args, **kwargs)
    727 
    728 

/Library/anaconda/envs/py36/lib/python3.6/site-packages/mne/connectivity/spectral.py in spectral_connectivity(data, method, indices, sfreq, mode, fmin, fmax, fskip, faverage, tmin, tmax, mt_bandwidth, mt_adaptive, mt_low_bias, cwt_frequencies, cwt_n_cycles, block_size, n_jobs, verbose)
    877     # compute final connectivity scores
    878     con = list()
--> 879     for method, n_args in zip(con_methods, n_comp_args):
    880         # future estimators will need to be handled here
    881         if n_args == 3:

UnboundLocalError: local variable 'con_methods' referenced before assignment
```

</details>

&nbsp;

It seemed to me like this is a bug, so I thought I'd report it.

I am using mne 0.15.dev0 (on the latest github commit), but this also occurred with 0.14. I'm on Mac OS Sierra.

### PR DESCRIPTION
**Pull request by mmagnuski**
fixes #4364 
I can add a test later.

### CODE REVIEW

**Comment by mmagnuski** [comment]
I've added a test but it sits in the big, slow tests of spectral submodule, while the added test it is very light - I can move it to a separate test function.

**Comment by agramfort** [comment]
LGTM

please update what's new bug entry

**Comment by mmagnuski** [comment]
Ok, I'll do that later today.

**Comment by mmagnuski** [comment]
@agramfort 
whatsnew updated, all should be ready here

File: mne/connectivity/spectral.py [hunk 1]
```diff
@@ -469,6 +469,8 @@ def _get_n_epochs(epochs, n):
         if len(epochs_out) >= n:
             yield epochs_out
             epochs_out = list()
+    if 0 < len(epochs_out) < n:
```
Comment (jaeilepp) [inline mne/connectivity/spectral.py]: Can you also include case ``len(epochs_out) == 0``? It causes some joblib versions to hang forever. Fixed in https://github.com/mne-tools/mne-python/pull/4278, but apparently got lost somewhere.

Comment (jaeilepp) [inline mne/connectivity/spectral.py]: No, It's actually the other way round. LGTM.

**Comment by jaeilepp** [comment]
Thanks @mmagnuski

**Comment by mmagnuski** [comment]
Thanks @agramfort and @jaeilepp ! 🚀

**Pull request merged**